### ISSUE TITLE
Delay time should be displayed.

### ISSUE THREAD

**Issue by maormagori** [guytepper/better-rail#57]
Love the app.
A few things are missing though. The most important one in my opinion is the delay time.

As you can see, the official site tells you the train is delayed and by how much:

![Screenshot_20210728-233126_Chrome.jpg](https://user-images.githubusercontent.com/24867863/127392760-c661c5a3-22da-4687-85fb-986377a22d89.jpg)

But the app doesn't

![Screenshot_20210728-233135_Better Rail.jpg](https://user-images.githubusercontent.com/24867863/127392901-e08245fb-6077-4802-986b-d54d382cc0ff.jpg)

**Comment by guytepper** [comment]
Hey, thanks for letting me know about that. 
I had no idea that the website displays delay times, and I couldn't find the property in the API that indicate a delay. 

Do you know any routes that have frequent delays so I could investigate how it gets reported?

**Comment by maormagori** [comment]
@guytepper 
The 23:00 trains from Tel Aviv University to Tel Aviv Hahagana always get delayed if you'd like to investigate:

![2021-07-29 23_27_57-רכבת ישראל תוצאות חיפוש](https://user-images.githubusercontent.com/24867863/127564842-47855878-5cb4-4939-a6ae-ef7e41d76a70.png)

Here's the GetRoutes API call at the time:

<details>
<summary><h3>GetRoutes.json</h3></summary>

```JSON
{
    "MessageType":2,
    "Message":null,
    "Data":{
        "Error":null,
        "Details":{
            "Origin":"3600",
            "Destination":"4900",
            "Date":"29/07/2021 00:00:00",
            "SugKav":"P",
            "Hour":"0"
        },
        "StartIndex":0,
        "Routes":[
            {
                "Train":[
                    {
                        "Trainno":"783",
                        "OrignStation":"3600",
                        "DestinationStation":"4900",
                        "ArrivalTime":"29/07/2021 23:19:00",
                        "DepartureTime":"29/07/2021 23:05:00",
                        "StopStations":[
                            {
                                "StationId":"3700",
                                "ArrivalTime":"29/07/2021 23:10:00",
                                "DepartureTime":"29/07/2021 23:12:00",
                                "Platform":"3"
                            },
                            {
                                "StationId":"4600",
                                "ArrivalTime":"29/07/2021 23:14:00",
                                "DepartureTime":"29/07/2021 23:14:00",
                                "Platform":"2"
                            }
                        ],
                        "LineNumber":"35000680",
                        "Route":"21",
                        "Midnight":false,
                        "Handicap":true,
                        "DirectTrain":true,
                        "TrainParvariBenironi":null,
                        "ReservedSeat":false,
                        "Platform":"2",
                        "DestPlatform":"3",
                        "IsFullTrain":false
                    }
                ],
                "IsExchange":false,
                "EstTime":"00:14:00"
            },
            {
                "Train":[
                    {
                        "Trainno":"135",
                        "OrignStation":"3600",
                        "DestinationStation":"4900",
                        "ArrivalTime":"29/07/2021 23:31:00",
                        "DepartureTime":"29/07/2021 23:17:00",
                        "StopStations":[
                            {
                                "StationId":"3700",
                                "ArrivalTime":"29/07/2021 23:22:00",
                                "DepartureTime":"29/07/2021 23:24:00",
                                "Platform":"3"
                            },
                            {
                                "StationId":"4600",
                                "ArrivalTime":"29/07/2021 23:26:00",
                                "DepartureTime":"29/07/2021 23:26:00",
                                "Platform":"2"
                            }
                        ],
                        "LineNumber":"1",
                        "Route":"1",
                        "Midnight":false,
                        "Handicap":true,
                        "DirectTrain":false,
                        "TrainParvariBenironi":null,
                        "ReservedSeat":false,
                        "Platform":"2",
                        "DestPlatform":"3",
                        "IsFullTrain":false
                    }
                ],
                "IsExchange":false,
                "EstTime":"00:14:00"
            },
            {
                "Train":[
                    {
                        "Trainno":"8135",
                        "OrignStation":"3600",
                        "DestinationStation":"4900",
                        "ArrivalTime":"29/07/2021 23:37:00",
                        "DepartureTime":"29/07/2021 23:23:00",
                        "StopStations":[
                            {
                                "StationId":"3700",
                                "ArrivalTime":"29/07/2021 23:28:00",
                                "DepartureTime":"29/07/2021 23:30:00",
                                "Platform":"3"
                            },
                            {
                                "StationId":"4600",
                                "ArrivalTime":"29/07/2021 23:32:00",
                                "DepartureTime":"29/07/2021 23:32:00",
                                "Platform":"2"
                            }
                        ],
                        "LineNumber":"2",
                        "Route":"1",
                        "Midnight":false,
                        "Handicap":true,
                        "DirectTrain":false,
                        "TrainParvariBenironi":null,
                        "ReservedSeat":false,
                        "Platform":"2",
                        "DestPlatform":"3",
                        "IsFullTrain":false
                    }
                ],
                "IsExchange":false,
                "EstTime":"00:14:00"
            }
        ],
        "BeforeRoutes":null,
        "Rates":[
            
        ],
        "Delays":[
            {
                "Station":"3100",
                "Date":"20210729",
                "Time":"232314",
                "Train":"135",
                "Min":"0013"
            },
            {
                "Station":"3100",
                "Date":"20210729",
                "Time":"232314",
                "Train":"8135",
                "Min":"0016"
            }
        ],
        "TrainPositions":[
            {
                "CurrentStation":3100,
                "DifMin":13,
                "NextStation":3300,
                "TrainNumber":135,
                "DifType":"DELAY",
                "TotalKronot":"007",
                "SugNayad":"DD",
                "StopPoint":"ג"
            },
            {
                "CurrentStation":8600,
                "DifMin":5,
                "NextStation":680,
                "TrainNumber":781,
                "DifType":"DELAY",
                "TotalKronot":"006",
                "SugNayad":"DD",
                "StopPoint":"ב"
            },
            {
                "CurrentStation":4900,
                "DifMin":3,
                "NextStation":0,
                "TrainNumber":783,
                "DifType":"DELAY",
                "TotalKronot":"006",
                "SugNayad":"DD",
                "StopPoint":"ב"
            },
            {
                "CurrentStation":3100,
                "DifMin":16,
                "NextStation":0,
                "TrainNumber":8135,
                "DifType":"DELAY",
                "TotalKronot":"008",
                "SugNayad":"DD",
                "StopPoint":"ד"
            }
        ],
        "Omasim":[ ]
    }
}
```
</details>

I left `Omasim` empty to keep it readable. As you can see the `Data.Delays` property is populated and `Min` is the amount of the delay. You can also see that `TrainPosition` also has the delay time in the `DifMin` property.

Btw, is the train position planned to be incorporated in the current layout?

**Comment by guytepper** [comment]
Thanks for the detailed response 🙏
I'm almost certain those are new additions to the API, I haven't noticed them initially. 

I'll try to push to add the delay information on the next major version. 
The `TrainPosition` is something I'd also love to add, but I'll have to understand more how it works before that. Have you noticed it implemented somewhere in Israel Railways web/mobile apps?

**Comment by guytepper** [comment]
Closing this in favor of #61

**Comment by maormagori** [comment]
@guytepper 
The design idea looks great!
Sorry for not coming back with an answer about implementing the `trainPosition` , the website is'nt really helping much.

**Comment by guytepper** [comment]
Thank you!

I ended up finding an implementation of the train position in their website; but it seems quite clunky, not really accurate and I'm not sure if it'd be useful for most users. It's not like a bus, that having it's location can provide actual value with deciding which route to take and how much time should the passenger wait.

**Comment by guytepper** [comment]
I've finished adding the delay times to the list screen, but I'm unsure if it works correctly. 
@maormagori would you like to give me feedback on this? if so, send me your email (your play store account address) to `feedback@better-rail.co.il` and I'll give you access to the beta version :)

**Comment by maormagori** [comment]
Sure!
I'll send right away.

**Comment by guytepper** [comment]
This is now released publicly in [v1.6.0](https://github.com/guytepper/better-rail/releases/tag/v1.6.0). 
Many thanks for the heads up on this feature 🙏
